src/index.js

```diff
@@ -10,45 +11,154 @@ class Todo {
   }
 }
 
-let todos = [
-  new Todo('Hello world 1', false, 0),
-  new Todo('Hello world 2', false, 1),
-  new Todo('Hello world 5', false, 4),
-  new Todo('Hello world 3', false, 2),
-  new Todo('Hello world 4', false, 3),
-];
+let todos = [];
 
-function populate() {
-  todos.sort((a, b) => (a.index > b.index ? 1 : -1));
-  todos.forEach((todo) => {
-    const li = document.createElement('li');
-    li.innerHTML = `
-    <div class="flex">
+function createTodoElement(todo) {
+  const li = document.createElement('li');
+  li.innerHTML = `
+    <div class="flex todo-element">
       <div>
           <input type="checkbox" class="checkbox" 
           ${todo.completed ? 'checked' : ''}>
           <span>${todo.description}</span>
       </div>
-      <span class="material-icons">
+      <span class="material-icons edit-icon" style="cursor: pointer">
           more_vert
       </span>
     </div>
     <hr>`;
 
-    button.parentElement.insertBefore(li, button);
+  return li;
+}
+
+function createReplaceTodoElement(todo) {
+  const html = `
+  <div>
+    <input type="checkbox" class="checkbox" 
+    ${todo.completed ? 'checked' : ''}>
+    <span>${todo.description}</span>
+  </div>
+  <span class="material-icons edit-icon" style=" cursor: pointer">
+      more_vert
+  </span>
+    `;
+
+  return html;
+}
+
+function addTodoElement(todo) {
+  const li = createTodoElement(todo);
+  button.parentElement.insertBefore(li, button);
+}
+
+function populate() {
+  todos.sort((a, b) => (a.index > b.index ? 1 : -1));
+  todos.forEach((todo) => {
+    addTodoElement(todo);
   });
 }
 
 function saveTodosLocally() {
   localStorage.setItem('todos', JSON.stringify(todos));
 }
 
-function addEventsToCheckboxes() {
+function createReplaceTodoElementForCompletedTask(todo) {
+  const html = `
+  
+  <div>
+  <span class="material-icons edit-icon" style=" cursor: pointer; color: green">
+      done
+  </span>
+    <strike><span>${todo.description}</span></strike>
+  </div>
+  <span class="material-icons edit-icon" style=" cursor: pointer">
+      more_vert
+  </span>
+    `;
+
+  return html;
+}
+
+function changeElementToCompleted(index) {
+  update(todos[index]);
+  saveTodosLocally();
+  if (todos[index].completed) {
+    const completedElement = createReplaceTodoElementForCompletedTask(todos[index]);
+    const todoElements = document.querySelectorAll('.todo-element');
+    todoElements[index].innerHTML = completedElement;
+  }
+}
```

Comment:
- [ ] [OPTIONAL] I think it could be good to toggle the complete state since for now we can only mark a completed but we can not reverse the changes or take it back to the initial state (completed: false)